include/net/nrf_cloud_pgps.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_PGPS_H_
+#define NRF_CLOUD_PGPS_H_
+
+/** @file nrf_cloud_pgps.h
+ * @brief Module to provide nRF Cloud P-GPS support to nRF9160 SiP.
+ */
+
+#include <zephyr.h>
+#include <drivers/gps.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup nrf_cloud_pgps nRF Cloud P-GPS
+ * @{
+ */
+
+struct nrf_cloud_pgps_prediction;
+
+/** @brief P-GPS request type */
+struct gps_pgps_request {
+	uint16_t prediction_count;
+	uint16_t prediction_period_min;
+	uint16_t gps_day;
+	uint32_t gps_time_of_day;
+} __packed;
+
+#define EAPPROXIMATE	8000 /* current time unknown; using first prediction */
+#define ELOADING	8001 /* not found but loading in progress */
+#define NRF_CLOUD_PGPS_EMPTY_EPHEM_HEALTH (0xff)
+
+enum nrf_cloud_pgps_event {
+	PGPS_EVT_INIT,
+	PGPS_EVT_UNAVAILABLE,
+	PGPS_EVT_LOADING,
+	PGPS_EVT_AVAILABLE,
+	PGPS_EVT_READY
+};
+
+typedef void (*pgps_event_handler_t)(enum nrf_cloud_pgps_event event,
+				     struct nrf_cloud_pgps_prediction *p);
+
+/**@brief Update storage of the most recent known location, in modem-specific
+ * normalized format (int32_t).
+ * Current time is also stored.
+ * Normalization:
+ *   (latitude / 90.0) * (1 << 23)
+ *   (longitude / 360.0) * (1 << 24)
```

Comment:
I'll verify the formula.

Reply:
It matches section "3.3.13.7 Location assistance" of the GNSS RPC API documentation.  The only slight difference is that I use  2^24/360 instead of 2^23/180 as the conversion factor for longitude.

Reply:
Sorry, you are right. I got confused by the bitshifts, but of course those are the same as the power of two.